[docs] Add an example for mocking cross-domain responses

VasilyStrelyaev opened the pull request:
\cc @miherlosev 

closes #3016

miherlosev commented on docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md:
```diff
@@ -12,6 +12,8 @@ Mocking is useful when the tested app uses infrastructure that is difficult to d
 * [The respond Method](#the-respond-method)
   * [A Custom Response Function](#a-custom-response-function)
+* [Examples](#examples)
+  * [Mocking Cross-Domain Responses](#mocking-cross-domain-responses)
```
`[Mocking Cross-Domain Requests](#mocking-cross-domain-requests)`

VasilyStrelyaev replied:
well, the topic is called `Mocking HTTP Responses` so it says we're mocking responses not requests.

VasilyStrelyaev replied:
Is the topic title wrong?

miherlosev replied:
As I know, for this case the phrase `Mocking http requests` is more used than `Mocking http responses'.
See module descriptions in https://www.npmjs.com/search?q=mocking%20requests.

miherlosev commented on docs/articles/documentation/test-api/intercepting-http-requests/mocking-http-responses.md:
```diff
@@ -117,3 +119,18 @@ Property | Type | Description
+
+### Mocking Cross-Domain Responses
+
+When you mock a cross-domain response (a response to a request to a different domain), set the response's `access-control-allow-origin` header to the tested site's domain (or an asterisk `*`) to pass [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) validation.
```
`When you mock a cross-domain requests, ...`.

Cross-domain request is a common term. You don't need explain this.

There are two most popular cases:
* if 'access-control-allow-credentials' is specified, then you should  setup an exact domain for 'access-control-allow-origin' header. With `*` specified, the CORS will fail in this case.
* if 'access-control-allow-credentials' is not specified, then you can specify the exact domain or `*`.

VasilyStrelyaev commented:
Renamed the topic, reflected changes regarding 'mock requests not responses' and added info about `access-control-allow-credentials`.

Hope everything is correct now.
